### DISCUSSION TITLE
STUN, etc

### DISCUSSION THREAD

**Discussion by bjtftw** [PurpleI2P/i2pd#1978] [Ideas]
I'm curious if I2P network would benefit from implementing something like STUN/ICE/etc. connection techniques for users behind NAT (often symmetric CGNAT) so they could be reachable from outside thanks to NAT hole punching and having one port fully open (reachable) from Internet?

I'm guessing that most I2PD (and I2P generally) users are unreachable due to NATs, they can start new connections to others but others can't initiate ones with them while NAT without port redirection won't allow new incoming connections.

Or am I misleading something and I2P don't need this and is working like VPN where all once connected users (even those behind NAT) can reach others with internal VPN IP address space?

**Comment by Vort** [comment]
It is already implemented, but works poorly because of bugs.

**Reply by bjtftw** [reply]
Don't you know if there are any plans to repair this in a near future or it's seen as low importance bug?

**Reply by Vort** [reply]
I don't know if there are any plans at all.
What I can say is that I told developers many times about one of the problems with NAT (`U`) nodes - wrong peer test results.
